emuu-client/src/components/videoPage/videoPage.js

```diff
@@ -0,0 +1,172 @@
+import { useState, useEffect } from "react";
+import React from "react";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Checkbox from "@material-ui/core/Checkbox";
+import Favorite from "@material-ui/icons/Favorite";
+import FavoriteBorder from "@material-ui/icons/FavoriteBorder";
+import Button from "react-bootstrap/Button";
+import TextField from "@mui/material/TextField";
+import { db } from "../../Firebase.js";
+import {
+  getDocs,
+  getDoc,
+  collection,
+  doc,
+  where,
+  query,
+  updateDoc,
+  increment,
+  arrayUnion,
+  arrayRemove,
+} from "firebase/firestore";
+
+function Video({ video, setVideo }) {
+  const displayName = localStorage.getItem("displayName");
+  const [checked, setChecked] = useState(false);
+  function checkLiked() {
+    let liked = video?.usersThatLiked?.includes(displayName); //check if there is a video and if there are users that liked stored
+    if (liked) {
+      setChecked(true);
+    } else {
+      setChecked(false);
+    }
+  }
+
+  useEffect(async () => {
+    if (video) {
+      localStorage.setItem("video", JSON.stringify(video));
+    }
+    if (localStorage.getItem("video")) {
+      let video = JSON.parse(localStorage.getItem("video"));
+      const collectionRef = collection(db, "Videos");
+      const queryData = await query(
+        collectionRef,
+        where("VideoUrl", "==", video.VideoUrl)
+      );
+      const _doc = await getDocs(queryData);
+      let id = "";
+      _doc.forEach((doc) => (id = doc.id));
+      const videoRef = doc(db, "Videos", id);
+      setVideo((await getDoc(videoRef)).data());
+      await updateDoc(videoRef, { Views: increment(1) });
+    }
+    if (!video && !localStorage.getItem("video")) {
+      //if there's no video on this page, redirect to home
```

Comment:
I think a better alternative would be to send them to an error page that the video got deleted or something went wrong.